### ISSUE TITLE
Set the number of threads used in `test_multithread_stress` dynamically

### ISSUE THREAD

**Issue by tuliom** [libnxz/power-gzip#75]
On PowerVM, there is a restriction on the amount of VAS windows that can be opened simultaneously, restricting the amount of threads that can be started in `test_multithread_stress`.

With this change, we might also want to tune this value on PowerNV servers depending on the amount of engines and cores available at the system.

**Comment by mscastanho** [comment]
@tuliom does the kernel expose the max number of simultaneous VAS windows somewhere, or is there a way we can calculate it during run time?

**Comment by tuliom** [comment]
@mscastanho I don't think that's available yet.

**Comment by mscastanho** [comment]
I confirmed the kernel is not exposing that information yet. There was a [patch](https://patchwork.ozlabs.org/project/linuxppc-dev/patch/60176ad795219afbeaf51ad596af4bae710617b7.camel@linux.ibm.com/) recently, but it needed some changes and was dropped on subsequent versions of the patch set. While that feature doesn't land, I don't think there's a simple way to dynamically scale the test. I added a workaround on the PR above for the meantime.
